Sources/MessagingInApp/State/InAppMessageReducer.swift

```diff
@@ -49,16 +49,30 @@ private func reducer(action: InAppMessageAction, state: InAppMessageState) -> In
 
     case .displayMessage(let message):
         if let queueId = message.queueId {
+            // Do not add queueId to shownMessageQueueIds if the message is persistent
+            // As it should be considered shown only when it is dismissed via close action.
```

Comment:
I think we should not mix these two concepts, logging a view telling gist services that the message is shown vs keeping track of if the message is shown internally. 

Because technically the message is indeed shown to the user and we rely on it to ensure duplication doesn't happen.

Reply:
Discussed this on the call and moved the checks to `shouldTrackMessageShown`, which will now be used in both middlewares and reducer for consistency.